sys-boot/lilo/lilo-24.2-r3.ebuild

```diff
@@ -0,0 +1,197 @@
+		newins "${FILESDIR}"/lilo.conf lilo.conf.example
+
+		newconfd "${WORKDIR}"/dolilo/dolilo.conf.d dolilo.example
+
+		dodoc CHANGELOG* readme/README.* readme/INCOMPAT README
+		docinto samples ; dodoc sample/*
+	fi
+
+	# This we don't use
+	rm -r "${ED}/etc/initramfs" || die
+	# This should be in /usr/lib and it should have the .install suffix
+	dodir /usr/lib/kernel
+	for dir in postinst.d postrm.d; do
+		mv "${ED}/etc/kernel/${dir}" "${ED}/usr/lib/kernel/${dir}" || die
+		mv "${ED}/usr/lib/kernel/${dir}/zz-runlilo" "${ED}/usr/lib/kernel/${dir}/90-runlilo.install" || die
+	done
+	# Insert wrapper for systemd's kernel-install
+	exeinto /usr/lib/kernel/install.d
+	newexe - 90-runlilo.install <<-EOF
+		#!/bin/sh
+		if [ "\${1}" = "add" ]; then
+			exec "\$(dirname \${0})/../postinst.d/90-runlilo.install"
+		elif [ "\${1}" = "remove" ]; then
+			exec "\$(dirname \${0})/../postrm.d/90-runlilo.install"
+		fi
+	EOF
+}
+
+# Check whether LILO is installed
+# This function is from /usr/sbin/mkboot from debianutils, with copyright:
+#
+#   Debian GNU/Linux
+#   Copyright 1996-1997 Guy Maor <maor@debian.org>
+#
+# Modified for Gentoo for use with the lilo ebuild by:
+#   Martin Schlemmer <azarah@gentoo.org> (16 Mar 2003)
+#
+lilocheck() {
+	local bootpart=
+	local rootpart="$(mount | grep -v "tmpfs" | grep -v "rootfs" | grep "on / " | cut -f1 -d " ")"
+
+	echo
+	ebegin "Checking whether LILO can be safely updated"
+
+	if [ "$(whoami)" != "root" ]; then
+		eend 1
+		eerror "Only root can check for LILO"
+		return 1
+	fi
+
+	if [ -z "${rootpart}" ]; then
+		eend 1
+		eerror "Could not determine root partition"
+		return 1
+	fi
+
+	if [ ! -f /etc/lilo.conf -o ! -x /sbin/lilo ]; then
+		eend 1
+		eerror "No LILO configuration in place"
+		return 1
+	fi
+
+	bootpart="$(sed -n "s:^boot[ ]*=[ ]*\(.*\)[ ]*:\1:p" /etc/lilo.conf)"
+
+	if [ -z "${bootpart}" ]; then
+		# lilo defaults to current root when 'boot=' is not present
+		bootpart="${rootpart}"
+	fi
+
+	if ! dd if=${bootpart} ibs=16 count=1 2>&- | grep -q 'LILO'; then
+		eend 1
+		eerror "No LILO signature on ${bootpart}"
+		ewarn "Check your /etc/lilo.conf, or run /sbin/lilo by hand."
+		return 1
+	fi
+
+	if grep -q "^[[:space:]]*password[[:space:]]*=[[:space:]]*\"\"" /etc/lilo.conf; then
+		eend 1
+		eerror "Interactive password entry configured"
+		ewarn "Run /sbin/lilo -p by hand."
+		return 1
+	fi
+
+	einfo "LILO on ${bootpart} is safe to update"
+	eend 0
+	return 0
+}
+
+pkg_postinst() {
+	if [ ! -e "${ROOT}/boot/boot.b" -a ! -L "${ROOT}/boot/boot.b" ]; then
+		[ -f "${ROOT}/boot/boot-menu.b" ] && \
+			ln -snf boot-menu.b "${ROOT}/boot/boot.b"
+	fi
+
+	if [[ -z "${ROOT}" ]] && use !minimal; then
+		if lilocheck; then
+			einfo "Running DOLILO to complete the install"
+			# do not redirect to /dev/null because it may display some input
+			# prompt
+			/sbin/dolilo
+			if [ "$?" -ne 0 ]; then
+				eerror "You must manually configure and run LILO"
+			fi
+		fi
+		echo
+	fi
+	if use !minimal; then
```

Comment:
If you make the earlier `!` move change, do it here too, but again, fine if not.